File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_map<char, int>mp;
        int ans = 0;
        for(int high = 0, low = 0; high < s.size(); high++) {
            mp[s[high]]++;
            while(mp[s[high]] > 1) {
                mp[s[low]]--;
                low++;
            }
            ans = max(ans, high - low + 1);
        }
        return ans;
    }
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        HashMap<Character, Integer>mp = new HashMap<>();
        int ans = 0;
        for(int high = 0, low = 0; high < s.length(); high++) {
            mp.put(s.charAt(high), mp.getOrDefault(s.charAt(high), 0) + 1);
            while(mp.get(s.charAt(high)) > 1) {
                mp.put(s.charAt(low), mp.getOrDefault(s.charAt(low), 0) -1);
                low++;
            }
            ans = Math.max(ans, high - low + 1);
        }
        return ans;
    }
}